tests/by-util/test_mkdir.rs

```diff
@@ -151,3 +157,26 @@ fn test_mkdir_trailing_dot() {
     let result = scene2.cmd("ls").arg("-al").run();
     println!("ls dest {}", result.stdout_str());
 }
+
+#[test]
+#[cfg(not(windows))]
+fn test_umask_compliance() {
+    fn test_single_case(umask_set: mode_t) {
+        let (at, mut ucmd) = at_and_ucmd!();
+
+        let original_umask = unsafe { umask(umask_set) };
+
+        ucmd.arg(TEST_DIR12).succeeds();
+        let perms = at.metadata(TEST_DIR12).permissions().mode() as mode_t;
+
+        assert_eq!(perms, (!umask_set & 0o0777) + 0o40000); // before compare, add the set GUID, UID bits
+        unsafe {
+            umask(original_umask);
+        } // set umask back to original
+    }
+
+    for i in 0o0..0o027 {
+        // tests all permission combinations
```

Comment:
These are not all the possible permission settings. That's fine, let's just make sure the comment matches the code.